src/app/data-model/Decode.h

```diff
@@ -148,7 +151,10 @@ CHIP_ERROR Decode(TLV::TLVReader & reader, Optional<X> & x)
  *
  * Decodes a nullable value.
  */
-template <typename X>
+template <
+    typename X,
+    typename std::enable_if_t<
+        std::is_same<decltype(Decode(std::declval<TLV::TLVReader &>(), std::declval<X &>())), CHIP_ERROR>::value, X> * = nullptr>
```

Comment:
Again, why is this needed?

Reply:
The same above, we need SFINAE to gurantee only one candicate with the highest priority for every type.